plugins/woocommerce/src/Admin/Features/ProductBlockEditor/RedirectionController.php

```diff
@@ -155,6 +155,13 @@ protected function get_parsed_route(): array {
 		$path        = esc_url_raw( wp_unslash( $_GET['path'] ) );
 		$path_pieces = explode( '/', wp_parse_url( $path, PHP_URL_PATH ) );
 
+		if ( 2 !== count( $path_pieces ) ) {
```

Comment:
@bor0 , I did only code review here but in next lines I see
```php
'product_id' => 'product' === $path_pieces[1] ? absint( $path_pieces[2] ) : null,
```

which suggests that `count( $path_pieces )` could be more than 2

Reply:
Fixed in 0fac6af5de, thanks!